frontend/src/features/leases/detail/LeasePages/insurance/edit/EditContainer.tsx

```diff
@@ -0,0 +1,202 @@
+import { ILookupCode } from 'store/slices/lookupCodes/interfaces';
+import { withNameSpace } from 'utils/formUtils';
+
+import { useUpdateInsurance } from './hooks/useUpdateInsurance';
+import InsuranceForm from './InsuranceForm';
+import { FormInsurance, IUpdateFormInsurance } from './models';
+
+export interface InsuranceEditContainerProps {
+  leaseId: number;
+  insuranceList: IInsurance[];
+  insuranceTypes: ILookupCode[];
+  onCancel: () => void;
+  onSuccess: () => void;
+}
+
+const InsuranceEditContainer: React.FunctionComponent<InsuranceEditContainerProps> = ({
+  leaseId,
+  insuranceList,
+  insuranceTypes,
+  onSuccess,
+  onCancel,
+}) => {
+  const formikRef = useRef<FormikProps<IUpdateFormInsurance>>(null);
+
+  const handleOnChange = (e: any, codeType: any, arrayHelpers: any) => {
+    if (formikRef.current) {
+      let found = initialInsurances.findIndex(x => x.insuranceType.id === codeType.id);
+
+      if (e.target.checked) {
+        arrayHelpers.push(codeType.id);
+        //formikRef.current?.values.insurances.push(FormInsurance.createNew(codeType));
+        formikRef.current.values.insurances[found].isShown = true;
+      } else {
+        const idx = formikRef.current?.values.visibleTypes.indexOf(codeType.id + '');
+        arrayHelpers.remove(idx);
+        formikRef.current.values.insurances[found].isShown = false;
+      }
+    }
+
+    //let found = initialInsurances.find(x => x.insuranceType.id === codeType.id);
+  };
+
+  const { batchUpdateInsurances } = useUpdateInsurance();
+
+  const handleSave = async (lease: IBatchUpdateRequest<IInsurance>) => {
+    console.log('Submited:', lease);
+    const leaseResponse = await batchUpdateInsurances(leaseId, lease);
+    console.log(leaseResponse);
+
+    if (leaseResponse?.errorMessages.length === 0) {
+      onSuccess();
+    }
+  };
+
+  const handleCancel = () => {
+    onCancel();
+  };
+
+  const initialInsurances = insuranceTypes.map<FormInsurance>(x => {
+    let foundInsurance = insuranceList.find(i => i.insuranceType.id === x.id);
+    if (foundInsurance) {
+      return FormInsurance.createFromModel(foundInsurance);
+    } else {
+      return FormInsurance.createEmpty(x);
+    }
+  });
+
+  const initialTypes = insuranceList.map<string>(x => x.insuranceType.id);
+
+  const initialValues: IUpdateFormInsurance = {
+    insurances: initialInsurances,
+    visibleTypes: initialTypes,
+  };
+
+  return (
+    <Formik<IUpdateFormInsurance>
+      initialValues={initialValues}
+      onSubmit={(values: IUpdateFormInsurance, formikHelpers) => {
+        const updatedVals = values.insurances.reduce(
+          (accumulator: IEntryModification<IInsurance>[], entry: FormInsurance) => {
+            const existingEntry = initialValues.insurances.find(i => i.id === entry.id && !i.isNew);
+            console.log(entry, existingEntry);
+
+            if (existingEntry) {
+              if (!entry.isShown) {
+                accumulator.push({
```

Comment:
Was wondering how this works in the case of a local delete. Ie: if a user creates a new entry, and then removes it before saving it.

Reply:
It works since the isShown is what defines if it was removed from the checklist.